main.py

```diff
@@ -0,0 +1,58 @@
+import uvicorn
+from fastapi import FastAPI, Depends
+from sqlalchemy.orm import Session
+
+import crud
+import models
+import schemas
+from database import SessionLocal
+
+app = FastAPI()
+
+def get_db() -> Session:
+    db = SessionLocal()
+
+    try:
+        yield db
+    finally:
+        db.close()
+
+@app.get("/")
+def root():
+    return {"message": "Hello World"}
+
+
+@app.get("/authors/", response_model=list[schemas.Author])
+def read_authors(db: Session = Depends(get_db)):
+    return crud.get_all_authors(db)
+
+
+@app.post("/authors/", response_model=schemas.Author)
+def create_author(
+        author: schemas.AuthorCreate,
+        db: Session = Depends(get_db)
+):
+    return crud.create_author(db=db, author=author)
+
+
+@app.get("/books/", response_model=list[schemas.Book])
+def read_books(db: Session = Depends(get_db), skip: int = 0, limit: int = 10,):
+    return crud.get_all_books(db, skip=skip, limit=limit)
+
+@app.post("/books/", response_model=schemas.Book)
+def create_author(
+        book: schemas.BookCreate,
+        db: Session = Depends(get_db)
+):
+    return crud.create_book(db=db, book=book)
+
+@app.get("/books/{book_id}", response_model=schemas.Book)
+def read_book(book_id: int, db: Session = Depends(get_db)):
+    return crud.get_book(db=db, book_id=book_id)
+
+@app.get("/authors/{author_id}", response_model=list[schemas.Author])

```

Comment:
The response model for `read_author` should be `schemas.Author` instead of `list[schemas.Author]` since it returns a single author object.